pdfminer/layout.py

```diff
@@ -660,11 +657,13 @@ def key_obj(t):
                 group = LTTextGroupLRTB([obj1, obj2])
             plane.remove(obj1)
             plane.remove(obj2)
-            dists = [ (c,d,obj1,obj2) for (c,d,obj1,obj2) in dists
-                      if (obj1 in plane and obj2 in plane) ]
+            removed = {obj1, obj2}
+            to_remove = [ (c,d,obj1,obj2) for (c,d,obj1,obj2) in dists
```

Comment:
Why are elements being removed from the dists list? Is it because accumulating elements to be removed, and removing them is less expensive than adding elements to dists (That satisfy the condition ```if (obj1 in plane and obj2 in plane)```? 
Also, instead of using the list comprehension, you can loop through the elements in dists, check which among them satisfies the condition, and remove it in the same loop. Something like - 
```
for (c,d,obj1, obj2) in dists:
    if (obj1 in removed or obj2 in removed):
        dists.remove( (c,d,obj1, obj2))
````